File: 06_greatest_three/06_greatest_three.py
# python program to find greatest among three number

# method 1 - traditional method

try:
    a=eval(input("Enter first number :"))
    b=eval(input("Enter second number :"))
    c=eval(input("Enter third number :"))
    if a>b:
        if a>c:
            print(f'{a} is greatest')
        else:
            print(f"{c} is greatest")
    else:
        if b>c:
            print(f"{b} is greatest")
        else:
            print(f"{c} is greatest")
except:
    print("Enter valid numeric value!")

# method 2 - using ternary operator

try:
    a=eval(input("Enter first number :"))
    b=eval(input("Enter second number :"))
    c=eval(input("Enter third number :"))
    print(f'{a} is greatest' if a>b and a>c else f'{b} is greatest' if b>a and b>c else f'{c} is greatest')
except:
    print("Enter Valid numeric value!")

# method 3 - function based approach

def greatest_among_three(a, b, c):
    if (a > b and a > c):
        return a
    elif (b > a and b > c):
        return b
    else:
        return c
    
try:
    a = int(input("Enter first number : "))
    b = int(input("Enter second number : "))
    c = int(input("Enter third number : "))

    if (a == b == c):
        print("Given numbers are equal")
    else:
        result = greatest_among_three(a, b, c)
        print(f"{result} is greates among {a}, {b}, {c}")

except ValueError as e:
    print(f"Invalid input : {e}")

except Exception as e:
    print(f"Excetion caught : {e}")

File: 06_greatest_three/06_greatest_three.dart
import "dart:io";

// program to find greatest among the three numbers

int greatesAmongThree(int a, int b, int c){
  if( a > b && a > c){
    return a;
  }
  else if(b > a && b > c){
    return b;
  }
  else{
    return c;
  }
}

void main(){
  try{
  stdout.write("Enter First number : ");
  int a = int.parse(stdin.readLineSync()!);

  stdout.write("Enter second number : ");
  int b = int.parse(stdin.readLineSync()!);

  stdout.write("Enter third number : ");
  int c = int.parse(stdin.readLineSync()!);

  if (a == b && b == c){
    print("All number are equal");
  }
  else{
    int result = greatesAmongThree(a, b, c);
    print("$result is greatest among three numbers");
  }
  }
  on FormatException catch(e){
    print("Invalid input : $e");
  }
  on Exception catch(e){
    print("Exception caught : $e");
  }
  catch(e){
    print("Exception caught : $e");
  }
}

